### PR TITLE
Integration tests for clusterreosurces group

### PR DESCRIPTION
**Pull request by worryg0d**
This PR adds integrations for all resources of `clusterresources` group.

It adds creation test cases for the following resources:
* `AWSEncryptionKey`
* `AWSEndpointServicePrincipal`

It adds deletion test cases for the following resources:
* `AWSEncrpytionKey`
* `AWSEndpointServicePrincipal`
*  `AWSSecurityGroupFirewallRule`
* `AWSVPCPerring`
* `AzureVNetPeering`
* `ClusterNetworkFirewallRule`

### CODE REVIEW

**Review by testisnullus — APPROVED** [review]

File: controllers/clusterresources/awsencryptionkey_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/instaclustr/operator/apis/clusterresources/v1beta1"
+	"github.com/instaclustr/operator/controllers/tests"
+	"github.com/instaclustr/operator/pkg/instaclustr"
```
Comment (DoodgeMatvey) [inline controllers/clusterresources/awsencryptionkey_controller_test.go:29]: please sort imports

File: controllers/clusterresources/azurevnetpeering_controller_test.go [hunk 2]
```diff
@@ -21,51 +21,79 @@ import (
...
+			Eventually(func() error {
+				_, err := instaClient.GetPeeringStatus(id, instaclustr.AzurePeeringEndpoint)
+				return err
+			}, timeout, interval).Should(Equal(instaclustr.NotFound))
+
+			By("checking if the resource exists in k8s")
+			Eventually(func() bool {
+				var resource v1beta1.AWSEncryptionKey
+				return errors.IsNotFound(k8sClient.Get(ctx, key, &resource))
+			}, timeout, interval).Should(BeTrue())
```
Comment (DoodgeMatvey) [inline controllers/clusterresources/azurevnetpeering_controller_test.go:96]: Why did you check for `AWSEncryptionKey` In all your tests?

**Review by DoodgeMatvey — CHANGES_REQUESTED** [review]

File: controllers/clusterresources/azurevnetpeering_controller_test.go (hunk 2 above)
Comment (worryg0d) [reply controllers/clusterresources/azurevnetpeering_controller_test.go:96]: Oops, I've mistyped.

Comment (worryg0d) [reply controllers/clusterresources/azurevnetpeering_controller_test.go]: Fixed

File: controllers/clusterresources/awsencryptionkey_controller_test.go (hunk 1 above)
Comment (worryg0d) [reply controllers/clusterresources/awsencryptionkey_controller_test.go]: Fixed

**Review by ribaraka — APPROVED** [review]

**Review by tengu-alt — APPROVED** [review]

**Review by DoodgeMatvey — APPROVED** [review]

**Pull request merged**